infomapcog/distances.py

```diff
@@ -73,32 +73,57 @@ def needleman_wunsch(x, y, lodict={}, gop=-2.5, gep=-1.75):
     'lodict' must be a dictionary with all symbol pairs as keys and
     match scores as values, or a False value (including an empty
     dictionary) to denote (-1, 1) scores. gop and gep are gap
-    penalties for opening/extending a gap.
+    penalties for opening/extending a gap; alternatively, you can set
+    'gop' to None and provide element/gap alignment costs.
 
     Returns the alignment score and one optimal alignment.
+    
+    >>> needleman_wunsch("AAAAABBBB", "AACAABBCB")
+    (5.0, [('A', 'A'), ('A', 'A'), ('A', 'C'), ('A', 'A'), ('A', 'A'), ('B', 'B'), ('B', 'B'), ('B', 'C'), ('B', 'B')])
+    >>> needleman_wunsch("banana", "mancala", local=True)
+    (2.0, [('a', 'a'), ('n', 'n')])
+    >>> needleman_wunsch("abc", "t", lodict={('a', ''): 0, ('b', ''): -2, ('c', ''): -0.5}, gop=None)
+    (-1.5, [('a', ''), ('b', 't'), ('c', '')])
 
     """
     n, m = len(x), len(y)
     dp = np.zeros((n + 1, m + 1))
     pointers = np.zeros((n + 1, m + 1), np.int32)
-    for i in range(1, n + 1):
-        dp[i,0] = dp[i-1,0]+(gep if i>1 else gop)
-        pointers[i,0]=1
-    for j in range(1,m+1):
-        dp[0,j] = dp[0,j-1]+(gep if j>1 else gop)
-        pointers[0,j]=2
+    if not local:
+        for i in range(1, n + 1):
```

Comment:
Yes, that would work, but it would be slightly ugly because the effective indices are practically 1-based, whereas enumerate will give 0-based indices.